WalletWasabi/BranchNBound/SendCoinSelector.cs

```diff
@@ -0,0 +1,253 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.BranchNBound
+{
+	public class SendCoinSelector
+	{
+		private List<Money>? FinalCoins { get; set; }
+		private bool _optimizedCoinsFound = false;
+
+		private Money _costOfHeader = Money.Satoshis(0);
+		private Money _costPerOutput = Money.Satoshis(0);
+		private int _bnbTryLimit = 5;
+		private Random _random = new();
+		private Money[]? UtxoSorted { get; set; }
+
+		public bool TryBranchAndBound(List<Money> coins, Money target, ulong maxTolerance, ulong toleranceIncrement, out ulong tolerance, out List<Money> finalCoins)
```

Comment:
`out ulong tolerance`? Did I miss something?

Reply:
I use that for an `Assert.True` to see if the end result is indeed enough

Reply:
But we shouldn't change the algorithm in favor of a unit test.

Reply:
I do agree with that, fixing